ros2_batch_job/__main__.py

```diff
@@ -553,19 +446,30 @@ def need_package_from_pipy(pkg_name):
                 ['"%s"' % job.python, '-m', 'pip', 'install', '-U'] +
                 ['colcon/%s' % pkgname for pkgname, importname, version in colcon_packages],
                 shell=True)
+            print('# END SUBSECTION')
 
         colcon_script = which('colcon')
 
-        # Show what pip has
-        job.run(['"%s"' % job.python, '-m', 'pip', 'freeze', '--all'], shell=True)
-        print('# END SUBSECTION')
-
         # Fetch colcon mixins
         if args.colcon_mixin_url:
+            print('# BEGIN SUBSECTION: Fetch colcon mixins')
             true_cmd = 'VER>NUL' if sys.platform == 'win32' else 'true'
             job.run([colcon_script, 'mixin', 'remove', 'default', '||', true_cmd], shell=True)
             job.run([colcon_script, 'mixin', 'add', 'default', args.colcon_mixin_url], shell=True)
             job.run([colcon_script, 'mixin', 'update', 'default'], shell=True)
+            print('# END SUBSECTION')
+
+        print('# BEGIN SUBSECTION: Print python versions')
+        # Print setuptools version
+        job.run(['"%s"' % job.python, '-c', '"import setuptools; print(setuptools.__version__)"'],
+                shell=True)
```

Comment:
Is shell=True specifically needed here? It doesn't seem like it unless there's something about `job.python` which requires shell interpretation before execution.

Reply:
I copied this over from the old code, but I agree I don't think we need this.  Let me remove it and see what happens here.

Reply:
It turns out that we do need this; if you try to run jobs without `shell=True`, you get "FileNotFoundError: [WinError 2] The system cannot find the file specified".

I'm going to revert this back to having `shell=True` for now.